base/java-tools/src/com/netscape/cmstools/CMCRequest.java

```diff
@@ -1926,6 +1938,147 @@ private static PKIData constructDecryptedPopRequest(
         return pkidata;
     }
 
```

Comment:
I know this is just moving existing code, but I think this can be handled more easily with the recent try-with-resources feature of java. Just do something like this:

try (FileOutputStream os = new FileOutputStream(ofilename) ){
                cmcblob.encode(os);
                System.out.println("");
                System.out.println("");
                System.out.println("The CMC enrollment request in binary format is stored in " +
                        ofilename);
            } catch (IOException e) {
                System.out.println("CMCRequest:  unable to process file " + ofilename +
                        " for writing:\n" + e);
            }
    }

Reply:
And this eliminates the whole clean up the resource code, which is kind of odd in the original code.